dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -77,45 +89,368 @@ void TypeLookupService::get_type_dependencies_i(const TypeIdentifierSeq& type_id
     }
   }
 }
 
-void TypeLookupService::get_type_dependencies(const TypeIdentifierSeq& type_ids,
-  TypeIdentifierWithSizeSeq& dependencies) const
-{
-  ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
-  get_type_dependencies_i(type_ids, dependencies);
-}
-
 void TypeLookupService::add_type_objects_to_cache(const TypeIdentifierTypeObjectPairSeq& types)
 {
   ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
   for (unsigned i = 0; i < types.length(); ++i) {
-    const TypeMap::iterator it_type_id_with_size_seq = minimal_type_map_.find(types[i].type_identifier);
-    if (it_type_id_with_size_seq == minimal_type_map_.end()) {
-      minimal_type_map_.insert(std::make_pair(types[i].type_identifier, types[i].type_object));
+    const TypeMap::iterator pos = type_map_.find(types[i].type_identifier);
+    if (pos == type_map_.end()) {
+      type_map_.insert(std::make_pair(types[i].type_identifier, types[i].type_object));
     }
   }
 }
 
 void TypeLookupService::add(TypeMap::const_iterator begin, TypeMap::const_iterator end)
 {
   ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
-  minimal_type_map_.insert(begin, end);
+  type_map_.insert(begin, end);
 }
 
 void TypeLookupService::add(const TypeIdentifier& ti, const TypeObject& tobj)
 {
   ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
-  TypeMap::const_iterator it_type_id_with_size_seq = minimal_type_map_.find(ti);
-  if (it_type_id_with_size_seq == minimal_type_map_.end()) {
-    minimal_type_map_.insert(std::make_pair(ti, tobj));
+  TypeMap::const_iterator pos = type_map_.find(ti);
+  if (pos == type_map_.end()) {
+    type_map_.insert(std::make_pair(ti, tobj));
+  }
+}
+
+void TypeLookupService::update_type_identifier_map(const TypeIdentifierPairSeq& tid_pairs)
+{
+  for (ACE_CDR::ULong i = 0; i < tid_pairs.length(); ++i) {
+    const TypeIdentifierPair& pair = tid_pairs[i];
+    complete_to_minimal_ti_map_.insert(std::make_pair(pair.type_identifier1, pair.type_identifier2));
+  }
+}
+
+void TypeLookupService::cache_type_info(const DDS::BuiltinTopicKey_t& key,
+                                        const TypeInformation& type_info)
+{
+  ACE_GUARD(ACE_Thread_Mutex, g, mutex_);
+  if (type_info_map_.find(key) == type_info_map_.end()) {
+    type_info_map_.insert(std::make_pair(key, type_info));
+  }
+}
+
+const TypeInformation& TypeLookupService::get_type_info(const DDS::BuiltinTopicKey_t& key) const
+{
+  ACE_GUARD_RETURN(ACE_Thread_Mutex, g, mutex_, type_info_empty_);
+  const TypeInformationMap::const_iterator it = type_info_map_.find(key);
+  if (it != type_info_map_.end()) {
+    return it->second;
+  }
+  return type_info_empty_;
+}
+
+DCPS::String TypeLookupService::equivalence_hash_to_string(const EquivalenceHash& hash) const
+{
+  std::ostringstream out;
+  out << "(";
+  for (unsigned i = 0; i < sizeof(EquivalenceHash); ++i) {
+    out << hash[i];
+    if (i < sizeof(EquivalenceHash) - 1) {
+      out << ", ";
+    } else {
+      out << ")";
+    }
+  }
```

Comment:
@sonndinh It doesn't appear like you are printing it in hex in this for loop, is that what you wanted or should it be changed to hex?

Reply:
It's intentional to print it in decimals because those hash values are also printed in decimals in the generated code by the IDL compiler. If we print it in a different format here, it would be harder to debug in case errors happen.

Reply:
Unfortunate, but alright, that makes sense. Looking at[ the `opendds_idl` code for this though I see that it casts the value to an `int`](https://github.com/objectcomputing/OpenDDS/blob/8ca40e4480cd6584d2af9afe768be8cfc3c33f56/dds/idl/typeobject_generator.cpp#L403) before printing it. This should too because the value otherwise will almost certainly be printed as a `char`.  It'd be even better if the two functions shared the same code if you really want them to be linked like what you're saying.